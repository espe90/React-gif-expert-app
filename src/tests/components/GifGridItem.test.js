import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Testing GifGridItem Component', () => {
    let wrapper;

    const title = 'Title';
    const url = 'https://url.jpg';

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url}/>);
    })
    
    test('should create component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should send proptypes title', () => {
        const text = wrapper.find('p').text().trim();
        expect(text).toBe(title);
    })

    test('should send proptypes url', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should had classname animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBeTruthy();
    })
})
