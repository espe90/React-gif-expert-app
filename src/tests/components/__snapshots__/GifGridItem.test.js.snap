// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Component should create component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://url.jpg"
  />
  <p>
     
    Title
     
  </p>
</div>
`;
